xstream/src/java/com/thoughtworks/xstream/converters/extended/AtomicReferenceConverter.java

```diff
@@ -40,23 +40,29 @@ public boolean canConvert(final Class type) {
     }
 
     public void marshal(final Object source, final HierarchicalStreamWriter writer, final MarshallingContext context) {
-        final AtomicReference ref = (AtomicReference)source; 
-        writer.startNode(mapper.serializedMember(AtomicReference.class, "value"));
-
-        final Object object = ref.get();
-        final String name = mapper.serializedClass(object!= null ? object.getClass() : null);
-        writer.addAttribute(mapper.aliasForSystemAttribute("class"), name);
-        context.convertAnother(ref.get());
-        writer.endNode();
+        final AtomicReference ref = (AtomicReference)source;
+        if (ref.get() != null) {
+            writer.startNode(mapper.serializedMember(AtomicReference.class, "value"));
+
+            final Object object = ref.get();
+            final String name = mapper.serializedClass(object != null ? object.getClass() : null);
+            writer.addAttribute(mapper.aliasForSystemAttribute("class"), name);
+            context.convertAnother(ref.get());
+            writer.endNode();
+        }
```

Comment:
Why `ref.get()` is called 3 times in this method?
Also checking `object != null` is redundant. Because there is already condition `ref.get() != null`

```suggestion
        final Object object = ref.get();
        if (object != null) {
            writer.startNode(mapper.serializedMember(AtomicReference.class, "value"));

            final String name = mapper.serializedClass(object.getClass());
            writer.addAttribute(mapper.aliasForSystemAttribute("class"), name);
            context.convertAnother(object);
            writer.endNode();
        }
```